momento-sdk/src/main/java/momento/sdk/responses/auth/GenerateDisposableTokenResponse.java

```diff
@@ -0,0 +1,52 @@
+package momento.sdk.responses.auth;
+
+import com.google.gson.Gson;
+import java.util.HashMap;
+import java.util.Map;
+import momento.sdk.auth.accessControl.ExpiresAt;
+import momento.sdk.exceptions.SdkException;
+import momento.token._GenerateDisposableTokenResponse;
+
+public interface GenerateDisposableTokenResponse {
+
+  class Success implements GenerateDisposableTokenResponse {
+    private final String authToken;
+    private final String endpoint;
+    private final ExpiresAt expiresAt;
+
+    public Success(_GenerateDisposableTokenResponse response) {
+      Map<String, Object> jsonMap = new HashMap<>();
+      jsonMap.put("endpoint", response.getEndpoint());
+      jsonMap.put("api_key", response.getApiKey());
+      String jsonString = new Gson().toJson(jsonMap);
+      byte[] jsonBytes = jsonString.getBytes(java.nio.charset.StandardCharsets.UTF_8);
+      authToken = java.util.Base64.getEncoder().encodeToString(jsonBytes);
+      endpoint = response.getEndpoint();
+      expiresAt = ExpiresAt.fromEpoch((int) response.getValidUntil());
+    }
+
+    @Override
+    public String toString() {
+      int len = authToken.length();
+      return authToken.substring(0, 10) + "..." + authToken.substring(len - 10, len);
+    }
+
+    public String authToken() {
+      return authToken;
+    }
+
+    public String endpoint() {
+      return endpoint;
+    }
+
+    public ExpiresAt expiresAt() {
+      return expiresAt;
+    }
```

Comment:
these seem good to me now, but @nand4011 you should weigh in if you have any thoughts on the consistency of this response class with the existing response classes.